File: backend/app/schemas/document.py
from pydantic import BaseModel, Field


class DocumentUploadResponse(BaseModel):
    document_id: str
    filename: str
    stored_filename: str
    pages: int
    characters: int
    chunks: int
    vision_fallback_used: bool = False
    vision_pages_analyzed: list[int] = Field(default_factory=list)
    extraction_quality_warning: str | None = None


class DocumentChunkResponse(BaseModel):
    text: str
    page: int
    chunk_index: int


class DocumentDetailsResponse(BaseModel):
    document_id: str
    filename: str
    chunks: list[DocumentChunkResponse]

File: frontend/src/types/document.ts
export type UploadResponse = {
  document_id: string;
  filename: string;
  stored_filename: string;
  pages: number;
  characters: number;
  chunks: number;
  vision_fallback_used?: boolean;
  vision_pages_analyzed?: number[];
  extraction_quality_warning?: string | null;
};

export type Source = {
  text: string;
  page: number;
  chunk_index: number;
  score: number;
};

export type AskResponse = {
  answer: string;
  sources: Source[];
};